worker/src/body/body.rs

```diff
@@ -0,0 +1,144 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use bytes::Bytes;
+use futures_util::Stream;
+use http::HeaderMap;
+use http_body::Body as _;
+
+use crate::Error;
+
+type BoxBody = http_body::combinators::UnsyncBoxBody<Bytes, Error>;
+
+fn try_downcast<T, K>(k: K) -> Result<T, K>
+where
+    T: 'static,
+    K: Send + 'static,
+{
+    let mut k = Some(k);
+    if let Some(k) = <dyn std::any::Any>::downcast_mut::<Option<T>>(&mut k) {
+        Ok(k.take().unwrap())
+    } else {
+        Err(k.unwrap())
+    }
+}
+
+#[derive(Debug)]
+pub struct Body(Option<BoxBody>);
```

Comment:
I wonder if here we could make this Body an enum of like a normal body and a buffered body we'd get from a string/bytes. That way when we go turn this into a `web_sys::Response` we can just give the entire body to the response constructor and skip the streaming overhead.

Reply:
The overhead of the streaming should be minimal. `http_body::Body` does no buffering of its own. It simply returns the entire chunk of data if it's already available within a single poll (https://docs.rs/http-body/latest/http_body/struct.Full.html). It's turned into a single readablestream in JS-land, which again is instructed to not do any buffering of its own.